app/src/app/components/specmath-modal/modal.component.ts

```diff
@@ -48,10 +44,19 @@ export class ModalComponent implements OnInit {
     }
   }
 
+  submitName() {
```

Comment:
I'm thinking that newSpecName isn't necessary.  The user-entered spec name is always available at this.specNameFormControl.value, there's no need to copy it to another variable.  If you wanted easy access you could do:
get newSpecName: string { return this.specNameFormControl.value; }